lib/spack/spack/test/concretize.py

```diff
@@ -2360,6 +2362,26 @@ def test_select_lower_priority_package_from_repository_stack(
             assert s[name].concrete
             assert s[name].namespace == namespace
 
+    @pytest.mark.only_clingo("Old concretizer cannot reuse")
+    def test_reuse_specs_from_non_available_compilers(self, mutable_config, mutable_database):
+        """Tests that we can reuse specs with compilers that are not configured locally."""
+        # All the specs in the mutable DB have been compiled with %gcc@=10.2.1
```

Comment:
maybe also assert  this? `mutable_database.query_local(...)`?

Reply:
https://github.com/spack/spack/pull/43539/commits/d4bf4876aed5a801d9dc8b3ca3a74dacb7b9cfba